cmake/thirdparty/SetupHIP.cmake

```diff
@@ -15,24 +15,76 @@ find_package(HIP REQUIRED)
 message(STATUS "HIP version:      ${HIP_VERSION_STRING}")
 message(STATUS "HIP platform:     ${HIP_PLATFORM}")
 
-if("${HIP_PLATFORM}" STREQUAL "hcc" OR "${HIP_PLATFORM}" STREQUAL "amd")
-	set(HIP_RUNTIME_DEFINE "__HIP_PLATFORM_HCC__")
-elseif("${HIP_PLATFORM}" STREQUAL "nvcc")
-	set(HIP_RUNTIME_DEFINE "__HIP_PLATFORM_NVCC__")
+set(HIP_RUNTIME_INCLUDE_DIRS "${HIP_ROOT_DIR}/include")
+if(${HIP_PLATFORM} STREQUAL "hcc")
+	set(HIP_RUNTIME_DEFINES "-D__HIP_PLATFORM_HCC__")
+    set(HIP_RUNTIME_LIBRARIES "${HIP_ROOT_DIR}/lib/libhip_hcc.so")
+elseif(${HIP_PLATFORM} STREQUAL "clang" OR ${HIP_PLATFORM} STREQUAL "amd")
+    set(HIP_RUNTIME_DEFINES "-D__HIP_PLATFORM_HCC__;-D__HIP_ROCclr__")
+    set(HIP_RUNTIME_LIBRARIES "${HIP_ROOT_DIR}/lib/libamdhip64.so")
+elseif(${HIP_PLATFORM} STREQUAL "nvcc")
+    set(HIP_RUNTIME_DEFINES "-D__HIP_PLATFORM_NVCC__")
+    find_package(cudatoolkit)
+    set(HIP_RUNTIME_LIBRARIES "${CUDAToolkit_LIBRARY_DIR}/libcudart.so")
```

Comment:
My preference would be that this is via a find_library or via extraction of the library path from the `CUDA::cudart` target (assuming we don't need to worry about using `CUDA::cudart_static`)

Reply:
I'll try to implement the latter here. I'm hesitant to use `find_library` because for this case AMD's `hipcc` command forwards directly to `nvcc`, so we really need the full toolkit and not just the library.

Reply:
I ended up using find_library because trying to guess which property to extract from the `CUDA::cudart` target was taking too much time and looked prone to future breakage. And I realized it wasn't really any more robust.

I'm still not sure how well the CUDA parts of this work. What was there before, which used `find_package(cudatoolkit)`, was broken and no one seemed to notice. That makes me think HIP compilation for NVIDIA devices doesn't have any users right now. This new implementation will do the BLT smoke tests, but doesn't work in RAJA due to (I think) RAJA assuming that HIP presence implies rocPRIM presence.

If there are other clients that use BLT to compile HIP for NVIIDA devices it would be good to know.

Reply:
AFAIK I don't think anyone is using hip to target CUDA in production. If the smoke tests work that's good with me. I think most projects are like RAJA, and treat hip and rocm almost as synonyms.